dissect/volume/ddf/c_ddf.py

```diff
@@ -0,0 +1,325 @@
+#define DDF_VDCR_RAID1                  0x01
+#define DDF_VDCR_RAID3                  0x03
+#define DDF_VDCR_RAID4                  0x04
+#define DDF_VDCR_RAID5                  0x05
+#define DDF_VDCR_RAID6                  0x06
+#define DDF_VDCR_RAID1E                 0x11
+#define DDF_VDCR_SINGLE                 0x0F
+#define DDF_VDCR_CONCAT                 0x1F
+#define DDF_VDCR_RAID5E                 0x15
+#define DDF_VDCR_RAID5EE                0x25
+
+/* Raid Level Qualifier (RLQ) */
+#define DDF_VDCR_RAID0_SIMPLE           0x00
+#define DDF_VDCR_RAID1_SIMPLE           0x00
+#define DDF_VDCR_RAID1_MULTI            0x01
+#define DDF_VDCR_RAID3_0                0x00
+#define DDF_VDCR_RAID3_N                0x01
+#define DDF_VDCR_RAID4_0                0x00
+#define DDF_VDCR_RAID4_N                0x01
+#define DDF_VDCR_RAID5_0_RESTART        0x00
+#define DDF_VDCR_RAID6_0_RESTART        0x01
+#define DDF_VDCR_RAID5_N_RESTART        0x02
+#define DDF_VDCR_RAID5_N_CONTINUE       0x03
+
+#define DDF_VDCR_RAID1E_ADJACENT        0x00
+#define DDF_VDCR_RAID1E_OFFSET          0x01
+
+/* Secondary RAID Level (SRL) */
+#define DDF_VDCR_2STRIPED               0x00
+#define DDF_VDCR_2MIRRORED              0x01
+#define DDF_VDCR_2CONCAT                0x02
+#define DDF_VDCR_2SPANNED               0x03
+
+#define DDF_VDCR_CACHE_WB               (1 << 0)
+#define DDF_VDCR_CACHE_WB_ADAPTIVE      (1 << 1)
+#define DDF_VDCR_CACHE_RA               (1 << 2)
+#define DDF_VDCR_CACHE_RA_ADAPTIVE      (1 << 3)
+#define DDF_VDCR_CACHE_WCACHE_NOBATTERY (1 << 4)
+#define DDF_VDCR_CACHE_WCACHE_ALLOW     (1 << 5)
+#define DDF_VDCR_CACHE_RCACHE_ALLOW     (1 << 6)
+#define DDF_VDCR_CACHE_VENDOR           (1 << 7)
+
+struct VD_Configuration_Record {
+    uint32  Signature;
+    uint32  CRC;
+    char    VD_GUID[24];
+    uint32  Timestamp;
+    uint32  Sequence_Number;
+    char    Reserved1[24];
+    uint16  Primary_Element_Count;
+    uint8   Strip_Size;
+    uint8   Primary_RAID_Level;
+    uint8   RAID_Level_Qualifier;
+    uint8   Secondary_Element_Count;
+    uint8   Secondary_Element_Seq;
+    uint8   Secondary_RAID_Level;
+    uint64  Block_Count;
+    uint64  VD_Size;
+    uint16  Block_Size;                                     /* DDF 2.0 */
+    uint8   Rotate_Parity_Count;                            /* DDF 2.0 */
+    char    Reserved2[5];
+    char    Associated_Spares[32];
+    uint64  Cache_Policies_And_Parameters;
+    uint8   BG_Rate;
+    char    Reserved3[3];
+    uint8   MDF_Parity_Disks;                               /* DDF 2.0 */
+    uint16  MDF_Parity_Generator_Polynomial;                /* DDF 2.0 */
+    char    Reserved4[1];
+    uint8   MDF_Constant_Generation_Method;                 /* DDF 2.0 */
+    char    Reserved5[47];
+    char    Reserved6[192];
+    char    V0[32];
+    char    V1[32];
+    char    V2[16];
+    char    V3[16];
+    char    Vendor_Specific_Scratch_Space[32];
+    // uint32  Physical_Disk_Sequence[0];
```

Comment:
Are these unused?

Reply:
I also mentioned this on another PR. I leave these as commented to indicate that these are variable count entries. In C you'd just index into the raw memory, in Dissect we prefer to parse these in Python instead of cstruct.

Reply:
Maybe add a comment then, otherwise you'll keep getting this question ;-)